token/js/client/token.js

```diff
@@ -458,6 +481,91 @@ export class Token {
     return newAccount.publicKey;
   }
 
+  /**
+   * Create and initialize the associated account.
+   *
+   * This account may then be used as a `transfer()` or `approve()` destination
+   *
+   * @param owner User account that will own the new account
+   * @return Public key of the new associated account
+   */
+  async createAssociatedTokenAccount(owner: PublicKey): Promise<PublicKey> {
+    const associatedAddress = await Token.getAssociatedTokenAddress(
+      this.associatedProgramId,
+      this.programId,
+      this.publicKey,
+      owner,
+    );
+
+    return this.createAssociatedTokenAccountInternal(owner, associatedAddress);
+  }
+
+  async createAssociatedTokenAccountInternal(
+    owner: PublicKey,
+    associatedAddress: PublicKey,
+  ): Promise<PublicKey> {
+    await sendAndConfirmTransaction(
+      'CreateAssociatedTokenAccount',
+      this.connection,
+      new Transaction().add(
+        Token.createAssociatedTokenAccountInstruction(
+          this.associatedProgramId,
+          this.programId,
+          this.publicKey,
+          associatedAddress,
+          owner,
+          this.payer.publicKey,
+        ),
+      ),
+      this.payer,
+    );
+
+    return associatedAddress;
+  }
+
+  /**
+   * Get the associated account or create one if not found.
+   *
+   * This account may then be used as a `transfer()` or `approve()` destination
+   *
+   * @param owner User account that will own the new account
+   * @return Public key of the new associated account
+   */
+  async getOrCreateAssociatedAccountInfo(
+    owner: PublicKey,
+  ): Promise<AccountInfo> {
+    const associatedAddress = await Token.getAssociatedTokenAddress(
+      this.associatedProgramId,
+      this.programId,
+      this.publicKey,
+      owner,
+    );
+
+    // This is the optimum logic, considering TX fee, client-side computation,
+    // RPC roundtrips and guaranteed idempotent.
+    // Sadly we can't do this atomically;
+    try {
+      return await this.getAccountInfo(associatedAddress);
+    } catch (err) {
+      if (err.message == FAILED_TO_FIND_ACCOUNT) {
```

Comment:
I think it's worth handling the `Invalid account owner` error as well because otherwise, partners won't be able to create associated token accounts for addresses that receive small lamport transfers

Reply:
yeah, another reason we need to publish this helper in this library. Too hard to write correct. ;)